docs/book/deploying-zenml/zenml-cloud/cloud-system-architecture.md

```diff
@@ -0,0 +1,79 @@
+{% endhint %}
+
+## Scenario 1: Fully SaaS
+
+<div data-full-width="true">
+<figure><img src="../../.gitbook/assets/cloud_architecture_scenario_1.png" alt=""><figcaption><p>Scenario 1: Fully SaaS deployment</p></figcaption></figure>
+</div>
+
+In this scenario, all services are hosted on the ZenML Cloud infrastructure. The ZenML Cloud API and MLflow Tracking store secrets and data on the ZenML side. The Cloud API interact with the customer cloud for certain actions like triggering pipelines, deploying models etc.
+
+This scenario is meant for customers who want to get started quickly with ZenML Cloud and can to a certain extent expose ingress connections to an external SaaS provider. Importantly, the data and secrets are all stored on ZenML Cloud side.
+
+## Scenario 2a: Hybrid SaaS
+
+<div data-full-width="true">
+<figure><img src="../../.gitbook/assets/cloud_architecture_scenario_2a.png" alt=""><figcaption><p>Scenario 2a: Hybrid SaaS deployment</p></figcaption></figure>
+</div>
+
+This scenario builds on top of Scenario 1, with the exception being that the "Data Artifacts" move over the customer cloud. This can be set up quite easily by setting up an [artifact store](../../stacks-and-components/component-guide/artifact-stores/) with your MLOps stack.
+
+In this scenario, the ZenML Cloud only stores metadata while the customer cloud keeps potentially sensitive customer data. However, if the customer wants access to special ZenML Cloud features like viewing artifacts on the dashboard, triggering pipeline runs, etc then they have to also provide [connectors](../../stacks-and-components/auth-management/service-connectors-guide.md) which are ultimately stored as secrets on the ZenML side. Therefore, while this scenario keeps the data on customer side, it still gives access to ZenML Cloud to trigger actions via an ingress connection over the internet into the customer cloud.
+
+Note, that ZenML encrypts all customer secrets at transit and at rest.
+
+## Scenario 2b: Hybrid SaaS + Private Secret Store
+
+<div data-full-width="true">
+<figure><img src="../../.gitbook/assets/cloud_architecture_scenario_2b.png" alt=""><figcaption><p>Scenario 2a: Hybrid SaaS deployment + Private Secret Store</p></figcaption></figure>
+</div>
+
+This is almost the same as Scenario 2a, however in this the customer brings their own secret store to keep sensitive information on customer side. This allows users to use the [secrets API](../../user-guide/advanced-guide/secret-management/) and ensure their secrets do not leave their environment.
+
+The ZenML secret store still exists, and this is again to enable customers to create [connectors](../../stacks-and-components/auth-management/service-connectors-guide.md) to allow ZenML Cloud to triggers actions on their environment.
```

Comment:
Just to be clear, the secret store will be registered as a stack comp and the creds to access it will be with the server?

Reply:
No, this is more of a "global" thing like Repositories. We have yet to flesh out details

Reply:
@wjayesh I imagine you'll have the option to configure a secondary secrets store when you deploy a ZenML tenant, which is located customer side and to which clients will connect to directly, but this still needs to be fleshed out, as Hamza says. The main difference is the ZenML clients will connect **directly** to the secondary secrets store, which means they will need implicit access.

Reply:
I see, this makes sense. I added this option to register a secrets store through helm in the PRD.